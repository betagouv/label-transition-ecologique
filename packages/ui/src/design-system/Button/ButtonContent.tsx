import {Icon, IconSize} from '@design-system/Icon';
import {buttonThemeClassnames} from './theme';
import {ButtonContentProps, ButtonSize} from './types';

const getIconSize = (size: ButtonSize): IconSize | undefined => {
  // Les sizes des icÃ´nes ne matchent pas celles des boutons
  switch (size) {
    case 'xs':
      return 'sm';
    case 'sm':
      return 'md';
    case 'md':
      return 'md';
    case 'xl':
      return 'lg';
  }
};

/** Affiche le contenu d'un bouton */
const ButtonContent = ({
  variant = 'primary',
  size = 'md',
  icon,
  children,
  disabled,
}: ButtonContentProps) => {
  const buttonState = disabled ? 'disabled' : 'default';

  return (
    <>
      {!!icon && (
        <Icon
          icon={icon}
          size={getIconSize(size)}
          className={buttonThemeClassnames[variant][buttonState].icon}
        />
      )}

      {!!children && children}
    </>
  );
};

export default ButtonContent;
