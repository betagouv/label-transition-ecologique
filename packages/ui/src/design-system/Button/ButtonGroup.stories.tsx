import {useState} from 'react';
import {Meta, StoryObj} from '@storybook/react';

import {ButtonGroup} from './ButtonGroup';

const meta: Meta<typeof ButtonGroup> = {
  component: ButtonGroup,
  args: {
    activeButtonId: '2',
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
  render: args => {
    const [activeId, setActiveId] = useState(args.activeButtonId);
    return (
      <ButtonGroup
        {...args}
        activeButtonId={activeId}
        buttons={args.buttons.map(button => ({
          ...button,
          onClick: () => setActiveId(button.id),
        }))}
      />
    );
  },
};

export default meta;

type Story = StoryObj<typeof ButtonGroup>;

export const Default: Story = {
  args: {},
};

export const Size: Story = {
  args: {
    size: 'xs',
  },
};

export const Disabled: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1', disabled: true},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
};

export const Deux: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
    ],
  },
};
