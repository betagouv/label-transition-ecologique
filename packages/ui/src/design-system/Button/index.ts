export * from './Button';
export * from './NotificationButton';
