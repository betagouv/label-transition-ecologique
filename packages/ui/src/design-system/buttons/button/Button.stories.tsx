import {useRef} from 'react';
import {Meta, StoryObj} from '@storybook/react';

import {Button} from './Button';
import DoubleCheckIcon from '../../../assets/DoubleCheckIcon';

const meta: Meta<typeof Button> = {
  title: 'Design System/Button',
  component: Button,
  argTypes: {
    variant: {
      control: {type: 'select'},
    },
    size: {
      control: {type: 'select'},
    },
    icon: {
      control: {type: 'text'},
    },
    iconPosition: {
      control: {type: 'select'},
    },
  },
};

export default meta;

type Story = StoryObj<typeof Button>;

/** Bouton par défaut, sans aucune props renseignée. */
export const Default: Story = {
  args: {
    children: 'Button',
  },
};

/** Bouton avec disabled à true. */
export const Disabled: Story = {
  args: {
    children: 'Disabled',
    disabled: true,
  },
};

/** Bouton avec icône Remix. */
export const WithRemixIcon: Story = {
  args: {
    children: 'Remix Icon',
    icon: 'leaf-line',
    iconPosition: 'left',
  },
};

/** Bouton avec icône SVG custom. */
export const WithCustomSVGIcon: Story = {
  args: {
    children: 'SVG Icon',
    icon: className => <DoubleCheckIcon className={className} />,
    iconPosition: 'left',
  },
};

/** Différentes valeurs pour la props size. */
export const Sizes: Story = {
  render: () => (
    <div
      className="grid gap-5 items-end"
      style={{gridTemplateColumns: 'repeat(4,fit-content(0))'}}
    >
      {/* Icon Buttons */}
      <Button size="xs" icon="leaf-line" />
      <Button size="sm" icon="leaf-line" />
      <Button size="md" icon="leaf-line" />
      <Button size="xl" icon="leaf-line" />

      {/* Buttons */}
      <Button size="xs">XSmall</Button>
      <Button size="sm">Small</Button>
      <Button size="md">Medium</Button>
      <Button size="xl">XLarge</Button>

      {/* Buttons with icons */}
      <Button size="xs" icon="leaf-line">
        XSmall
      </Button>
      <Button size="sm" icon="leaf-line">
        Small
      </Button>
      <Button size="md" icon="leaf-line">
        Medium
      </Button>
      <Button size="xl" icon="leaf-line">
        XLarge
      </Button>
    </div>
  ),
};

/** Bouton avec différents variants. */
export const Variants: Story = {
  parameters: {},
  render: () => (
    <div
      className="grid gap-5 items-end bg-grey-2 p-10"
      style={{gridTemplateColumns: 'repeat(5,fit-content(0))'}}
    >
      {/* Icon buttons */}
      <Button icon="leaf-line" variant="primary" />
      <Button icon="leaf-line" variant="secondary" />
      <Button icon="leaf-line" variant="outlined" />
      <Button icon="leaf-line" variant="white" />
      <Button icon="leaf-line" variant="grey" />

      {/* Disabled icon buttons */}
      <Button icon="leaf-line" variant="primary" disabled />
      <Button icon="leaf-line" variant="secondary" disabled />
      <Button icon="leaf-line" variant="outlined" disabled />
      <Button icon="leaf-line" variant="white" disabled />
      <Button icon="leaf-line" variant="grey" disabled />

      {/* Default buttons */}
      <Button variant="primary">Primary</Button>
      <Button variant="secondary">Secondary</Button>
      <Button variant="outlined">Outlined</Button>
      <Button variant="white">White</Button>
      <Button variant="grey">Grey</Button>

      {/* Disabled default buttons */}
      <Button variant="primary" disabled>
        Primary
      </Button>
      <Button variant="secondary" disabled>
        Secondary
      </Button>
      <Button variant="outlined" disabled>
        Outlined
      </Button>
      <Button variant="white" disabled>
        White
      </Button>
      <Button variant="grey" disabled>
        Grey
      </Button>

      {/* Buttons with icon */}
      <Button icon="leaf-line" iconPosition="left" variant="primary">
        Primary
      </Button>
      <Button icon="leaf-line" iconPosition="left" variant="secondary">
        Secondary
      </Button>
      <Button icon="leaf-line" iconPosition="left" variant="outlined">
        Outlined
      </Button>
      <Button icon="leaf-line" iconPosition="left" variant="white">
        White
      </Button>
      <Button icon="leaf-line" iconPosition="left" variant="grey">
        Grey
      </Button>

      {/* Disabled buttons with icon */}
      <Button icon="leaf-line" iconPosition="right" variant="primary" disabled>
        Primary
      </Button>
      <Button
        icon="leaf-line"
        iconPosition="right"
        variant="secondary"
        disabled
      >
        Secondary
      </Button>
      <Button icon="leaf-line" iconPosition="right" variant="outlined" disabled>
        Outlined
      </Button>
      <Button icon="leaf-line" iconPosition="right" variant="white" disabled>
        White
      </Button>
      <Button icon="leaf-line" iconPosition="right" variant="grey" disabled>
        Grey
      </Button>
    </div>
  ),
};

/** Boutons avec valeur href. */
export const AnchorButtons: Story = {
  render: () => (
    <div className="flex items-end gap-5">
      <Button href="https://territoiresentransitions.fr/">Internal link</Button>
      <Button
        href="https://territoiresentransitions.fr/"
        icon="leaf-line"
        variant="outlined"
      />
      <Button
        href="https://territoiresentransitions.fr/"
        variant="outlined"
        icon="leaf-line"
        disabled
      >
        Disabled link
      </Button>
      <Button href="https://territoiresentransitions.fr/" external>
        External link
      </Button>
      <Button href="https://territoiresentransitions.fr/" external />
    </div>
  ),
};

/** Customisation du bouton avec des classes ou des inline styles. */
export const WithRef: Story = {
  render: () => {
    const buttonRef = useRef();
    const anchorRef = useRef();
    const onButtonClick = () => console.log(buttonRef.current);
    const onAnchorClick = () => console.log(anchorRef.current);
    return (
      <div className="flex items-end gap-5">
        <Button ref={buttonRef} onClick={onButtonClick}>
          Button with ref
        </Button>
        <Button
          ref={anchorRef}
          onClick={onAnchorClick}
          href="https://territoiresentransitions.fr/"
          external
        >
          Anchor with ref
        </Button>
      </div>
    );
  },
};

/** Customisation du bouton avec des classes ou des inline styles. */
export const CustomStyles: Story = {
  render: () => (
    <div className="flex items-end gap-5">
      <Button className="!bg-success-3 !border-success-3 hover:!bg-success hover:!border-success">
        Custom Class
      </Button>
      <Button style={{fontStyle: 'italic', borderRadius: '2rem'}}>
        Custom Style
      </Button>
    </div>
  ),
};
