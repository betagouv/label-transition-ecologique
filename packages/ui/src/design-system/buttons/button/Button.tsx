import classNames from 'classnames';
import {buttonThemeClassnames} from '../utils';
import ButtonContent from './ButtonContent';
import {ButtonProps, isAnchorButton, isDefaultButton} from '../types';

/**
 * Composant bouton par défaut, ayant pour props toutes les props habituelles d'un button tag.
 * */
export const Button = (props: ButtonProps) => {
  const {
    children,
    disabled = false,
    variant = 'primary',
    size = 'md',
    className,
  } = props;

  const isIconButton = !children;
  const {text, background, border} = buttonThemeClassnames[variant];

  const buttonClassname = classNames(
    'rounded-lg w-fit border border-solid font-bold group',
    {
      // Format du bouton en fonction de la taille choisie
      'text-xs py-2': size === 'xs',
      'text-sm py-3': size === 'sm',
      'text-base py-3.5': size === 'md',
      'text-2xl py-5': size === 'xl',

      // Padding en fonction de la taille choisie
      'px-5':
        (size === 'xs' && !isIconButton) || (size === 'xl' && isIconButton),
      'px-6': size !== 'xs' && !isIconButton,

      // Padding sur les boutons ne contenant qu'une icône
      'px-2': size === 'xs' && isIconButton,
      'px-3': size === 'sm' && isIconButton,
      'px-3.5': size === 'md' && isIconButton,

      // Styles du curseur
      'cursor-pointer': !disabled,
      'cursor-not-allowed': disabled,
    },
    text,
    background,
    border
  );

  if (isDefaultButton(props)) {
    return (
      <button {...props} className={classNames(buttonClassname, className)}>
        <ButtonContent {...{...props, children, disabled, variant, size}} />
      </button>
    );
  } else if (isAnchorButton(props)) {
    const {external = false} = props;

    return (
      <a
        {...props}
        // bg-none permet d'effacer un style dsfr appliqué à la balise <a/>
        // after:hidden supprime l'icône external par défaut du dsfr
        className={classNames(
          'inline-block w-fit bg-none after:hidden',
          buttonClassname,
          className
        )}
        target={external ? '_blank' : props.target}
        rel={external ? 'noreferrer noopener' : props.rel}
        onClick={evt => {
          if (disabled) evt.preventDefault();
        }}
      >
        <ButtonContent
          {...{...props, children, disabled, variant, size}}
          external={external || props.target === '_blank'}
        />
      </a>
    );
  }
};
