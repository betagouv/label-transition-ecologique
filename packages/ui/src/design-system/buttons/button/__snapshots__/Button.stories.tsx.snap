// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/Button AnchorButtons smoke-test 1`] = `
<div class="flex items-end gap-5">
  <a href="https://territoiresentransitions.fr/"
     class="inline-block w-fit bg-none after:hidden rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8"
  >
    <div class="flex items-center gap-2">
      <div>
        Internal link
      </div>
    </div>
  </a>
  <a href="https://territoiresentransitions.fr/"
     class="inline-block w-fit bg-none after:hidden rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-primary-7 hover:!border-primary-8"
  >
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </a>
  <a href="https://territoiresentransitions.fr/"
     class="inline-block w-fit bg-none after:hidden rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-3 !bg-white !border-primary-4"
  >
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Disabled link
      </div>
    </div>
  </a>
  <a href="https://territoiresentransitions.fr/"
     class="inline-block w-fit bg-none after:hidden rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8"
     target="_blank"
     rel="noreferrer noopener"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-external-link-line font-normal">
      </div>
      <div>
        External link
      </div>
    </div>
  </a>
  <a href="https://territoiresentransitions.fr/"
     class="inline-block w-fit bg-none after:hidden rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8"
     target="_blank"
     rel="noreferrer noopener"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-external-link-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </a>
</div>
`;

exports[`Design System/Button CustomStyles smoke-test 1`] = `
<div class="flex items-end gap-5">
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 !bg-success-3 !border-success-3 hover:!bg-success hover:!border-success">
    <div class="flex items-center gap-2">
      <div>
        Custom Class
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8"
          style="font-style: italic; border-radius: 2rem;"
  >
    <div class="flex items-center gap-2">
      <div>
        Custom Style
      </div>
    </div>
  </button>
</div>
`;

exports[`Design System/Button Default smoke-test 1`] = `
<button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
  <div class="flex items-center gap-2">
    <div>
      Button
    </div>
  </div>
</button>
`;

exports[`Design System/Button Disabled smoke-test 1`] = `
<button disabled
        class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-3 !bg-primary-4 !border-primary-4"
>
  <div class="flex items-center gap-2">
    <div>
      Disabled
    </div>
  </div>
</button>
`;

exports[`Design System/Button Sizes smoke-test 1`] = `
<div class="grid gap-5 items-end"
     style="grid-template-columns: repeat(4, fit-content(0px));"
>
  <button class="rounded-lg w-fit border border-solid font-bold group text-xs py-2 px-2 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-1">
      <div class="ri-leaf-line font-normal h-4 w-4 text-base leading-4">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-sm py-3 px-3 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-1">
      <div class="ri-leaf-line font-normal h-5 w-5 text-xl leading-5">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-2xl py-5 px-5 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-8 w-8 text-3xl leading-8">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-xs py-2 px-5 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-1">
      <div>
        XSmall
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-sm py-3 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-1">
      <div>
        Small
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div>
        Medium
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-2xl py-5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div>
        XLarge
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-xs py-2 px-5 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-1">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        XSmall
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-sm py-3 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-1">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Small
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Medium
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-2xl py-5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        XLarge
      </div>
    </div>
  </button>
</div>
`;

exports[`Design System/Button Variants smoke-test 1`] = `
<div class="grid gap-5 items-end bg-grey-2 p-10"
     style="grid-template-columns: repeat(5, fit-content(0px));"
>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-white hover:!text-white bg-secondary-1 hover:!bg-secondary-2 border-secondary-1 hover:!border-secondary-2">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-pointer text-primary-7 hover:!text-primary-8 bg-grey-1 hover:!bg-grey-2 border-grey-4 hover:!border-grey-4">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-not-allowed !text-primary-3 !bg-primary-4 !border-primary-4"
  >
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-not-allowed !text-grey-1 !bg-secondary-2 !border-secondary-2"
  >
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-not-allowed !text-primary-3 !bg-white !border-primary-4"
  >
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-not-allowed !text-primary-4 !bg-white !border-white"
  >
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-3.5 cursor-not-allowed !text-primary-4 !bg-white !border-grey-2"
  >
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal h-6 w-6 text-2xl leading-6">
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div>
        Primary
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-secondary-1 hover:!bg-secondary-2 border-secondary-1 hover:!border-secondary-2">
    <div class="flex items-center gap-2">
      <div>
        Secondary
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div>
        Outlined
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1">
    <div class="flex items-center gap-2">
      <div>
        White
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-primary-7 hover:!text-primary-8 bg-grey-1 hover:!bg-grey-2 border-grey-4 hover:!border-grey-4">
    <div class="flex items-center gap-2">
      <div>
        Grey
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-3 !bg-primary-4 !border-primary-4"
  >
    <div class="flex items-center gap-2">
      <div>
        Primary
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-grey-1 !bg-secondary-2 !border-secondary-2"
  >
    <div class="flex items-center gap-2">
      <div>
        Secondary
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-3 !bg-white !border-primary-4"
  >
    <div class="flex items-center gap-2">
      <div>
        Outlined
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-4 !bg-white !border-white"
  >
    <div class="flex items-center gap-2">
      <div>
        White
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-4 !bg-white !border-grey-2"
  >
    <div class="flex items-center gap-2">
      <div>
        Grey
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Primary
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-secondary-1 hover:!bg-secondary-2 border-secondary-1 hover:!border-secondary-2">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Secondary
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Outlined
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        White
      </div>
    </div>
  </button>
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-primary-7 hover:!text-primary-8 bg-grey-1 hover:!bg-grey-2 border-grey-4 hover:!border-grey-4">
    <div class="flex items-center gap-2">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Grey
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-3 !bg-primary-4 !border-primary-4"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Primary
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-grey-1 !bg-secondary-2 !border-secondary-2"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Secondary
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-3 !bg-white !border-primary-4"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Outlined
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-4 !bg-white !border-white"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        White
      </div>
    </div>
  </button>
  <button disabled
          class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-not-allowed !text-primary-4 !bg-white !border-grey-2"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-leaf-line font-normal">
      </div>
      <div>
        Grey
      </div>
    </div>
  </button>
</div>
`;

exports[`Design System/Button WithCustomSVGIcon smoke-test 1`] = `
<button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
  <div class="flex items-center gap-2">
    <svg class="h-4 w-4 fill-white group-hover:fill-white"
         width="18"
         height="18"
         viewbox="0 0 24 24"
         fill="none"
         xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M11.602 13.7615L13.014 15.1735L21.48 6.70747L22.894 8.12147L13.014 18.0015L6.65 11.6375L8.064 10.2235L10.189 12.3485L11.602 13.7605V13.7615ZM11.604 10.9335L16.556 5.98047L17.966 7.39047L13.014 12.3435L11.604 10.9335ZM8.777 16.5885L7.364 18.0015L1 11.6375L2.414 10.2235L3.827 11.6365L3.826 11.6375L8.777 16.5885V16.5885Z">
      </path>
    </svg>
    <div>
      SVG Icon
    </div>
  </div>
</button>
`;

exports[`Design System/Button WithRef smoke-test 1`] = `
<div class="flex items-end gap-5">
  <button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
    <div class="flex items-center gap-2">
      <div>
        Button with ref
      </div>
    </div>
  </button>
  <a href="https://territoiresentransitions.fr/"
     class="inline-block w-fit bg-none after:hidden rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8"
     target="_blank"
     rel="noreferrer noopener"
  >
    <div class="flex items-center gap-2 flex-row-reverse">
      <div class="ri-external-link-line font-normal">
      </div>
      <div>
        Anchor with ref
      </div>
    </div>
  </a>
</div>
`;

exports[`Design System/Button WithRemixIcon smoke-test 1`] = `
<button class="rounded-lg w-fit border border-solid font-bold group text-base py-3.5 px-6 cursor-pointer text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8">
  <div class="flex items-center gap-2">
    <div class="ri-leaf-line font-normal">
    </div>
    <div>
      Remix Icon
    </div>
  </div>
</button>
`;
